apache2/ag_mdb/ag_mdb.cpp

```diff
@@ -321,6 +383,54 @@ int SHM_create(PTR_VOID* new_shm_base, const char* db_name, int db_name_length,
 #endif
 }
 
+/**
+ ** Destroy the shared memory of AG Memory Database 
+ ** @param dbm: The handler of the database.
+ ** return: AGMDB_SUCCESS if successfully destroy shared memory, 
+ **         AGMDB_ERROR if failed.
+ */
+int SHM_destroy(struct agmdb_handler *dbm) {
+    int rc;
+    if(dbm == NULL)
+        return AGMDB_ERROR_HANDLE_NULL;
+    /* Windows doesn't do anything */
+#ifndef _WIN32
+    rc = shmctl(dbm->linux_shm_id, IPC_RMID, 0);
+    if(rc == -1)
+        return AGMDB_ERROR_SHM_LINUX_DESTROY_FAIL;
+#endif
+    return AGMDB_SUCCESS;
+}
+
+/**
+ ** Close the shared memory of AG Memory Database
+ ** @param dbm: The handler of the database.
+ ** return: AGMDB_SUCCESS if successfully close shared memory,
+ **         AGMDB_ERROR if failed. 
+ */
+int SHM_close(struct agmdb_handler* dbm) {
```

Comment:
is there a reason for us to add this function, so under what circumstance, it needs the process to de-attach?